### DISCUSSION TITLE
Does PowerPlatform-DataverseServiceClient use the OData v2.0 endpoint?

### DISCUSSION THREAD

**Discussion by dboeckle** [microsoft/PowerPlatform-DataverseServiceClient#315] [Q&A] [answered]
When I run a program with the Nuget Microsoft.PowerPlatform.Dataverse.Client --version 1.0.9 in AzureHoste, it shows me in App Insights that Posts requests are made to the POST /XRMServices/2011/Organization.svc endpoint (see screenshot).	
 According to  [do-not-use-odata-v2-endpoint](https://learn.microsoft.com/en-us/power-apps/developer/model-driven-apps/best-practices/business-logic/do-not-use-odata-v2-endpoint) and  [OData v2.0 Service removal](https://powerapps.microsoft.com/en-us/blog/odata-v2-0-service-removal-date-announcement/) is a reference to Odata v2.0. 
Does this mean that I will not be able to use this Nuget package after 11/11/2022, or is really only Organization.svc and not OrganizationData.svc affected?

Please forgive if the question has already been answered I have not seen it yet

Answered in this issue https://github.com/microsoft/PowerPlatform-DataverseServiceClient/issues/310

Sorry
![AppInsights](https://user-images.githubusercontent.com/59045338/191013430-aebf0eb4-3d2a-49aa-9640-18952a8548d9.JPG)

**Comment by MattB-msft** [comment, marked as answer]
The dvService client is fully supported and is not affected by the odata 2.0 deprecation. 

Internally it uses the platform native api and the odata 4.0 api (we api)

**Reply by michaelmmcrae1** [reply]
Great to hear Matt.

A couple clarifications...

1.
Can you clarify which versions are SAFE, NOT using OData v2? All versions ever released of this package? You see, when we deploy apps, they are using whatever version from that time. Deployed Apps obviously don't update Nuget packages to later versions on their own. We need to know per version.

2.
What do you mean by "the platform native api"? That seems awfully vague when we need to be sure it's NOT using OData v2.

3.
Related... I can easily see this coming up again in 8 years when Microsoft announces deprecating "Web API"! and we will again be wondering. So it seems like you should be tracking internal API dependencies on EVERY version that is released. So we can just check your chart (Version | Internal API endpoint used) instead of asking here again.

**Reply by MattB-msft** [reply]
1) The CrmServiceClient and Dataverse ServiceClient have never used the OdataV2 API. you are safe to use any version of that. 
If your using Microsoft.Xrm.Client.dll (which we deprecated and removed from our distribution in 2016), that client does use OdataV2 and will break. 

2) Platform native API is what we call the "Organization API" or the XrmAPI.  It is surfaced as "[IOrganziationService](https://learn.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.iorganizationservice?view=dataverse-sdk-latest)" https://learn.microsoft.com/en-us/power-apps/developer/data-platform/org-service/overview

We use different transport protocols to surface this to various clients, the protocol's we are using are 'closed' and not intended for integration with non-shipped clients.

Publicly we surface the ODataV4 (standards based) WebAPI  for folks to integrate with generally, and also offer a way to invoke it from within our clients.  

3) for developers that roll their own clients, using our published API endpoints, yes. You will need to keep up with this and pay attention when we give notice.  In general, we give a year to 2 years notice on such a change. 

If you use our provided clients ( CrmServiceClient / Dataverse ServiceClient and such ), we work very hard to make any changes as transparent as generally this entails asking customers to update their supporting library's or, most recently, make a configuration change to authentication.
